query.go

```diff
@@ -574,3 +577,140 @@ func aurInfoPrint(names []string) ([]*rpc.Pkg, error) {
 
 	return info, nil
 }
+
+func aurPkgbuilds(names []string) ([]string, error) {
+	pkgbuilds := make([]string, 0, len(names))
+	var mux sync.Mutex
+	var wg sync.WaitGroup
+	var errs multierror.MultiError
+
+	makeRequest := func(name string) {
+		defer wg.Done()
+
+		values := url.Values{}
+		values.Set("h", name)
+
+		url := "https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?"
+
+		resp, err := http.Get(url + values.Encode())
+		if err != nil {
+			errs.Add(err)
+			return
+		}
+		defer resp.Body.Close()
+
+		body, readErr := ioutil.ReadAll(resp.Body)
+		pkgbuild := string(body)
+
+		if readErr != nil {
+			errs.Add(readErr)
+			return
+		}
+
+		if strings.Contains(pkgbuild,
+			"<div class='content'><div class='error'>Invalid branch: "+name+"</div>") {
+			errs.Add(fmt.Errorf("package \"%s\" was not found", name))
+			return
+		}
+
+		mux.Lock()
+		pkgbuilds = append(pkgbuilds, pkgbuild)
+		mux.Unlock()
+	}
+
+	for _, name := range names {
+		wg.Add(1)
+		go makeRequest(name)
+	}
+
+	wg.Wait()
+
+	if err := errs.Return(); err != nil {
+		return pkgbuilds, err
+	}
+
+	return pkgbuilds, nil
+}
+
+func repoPkgbuilds(names []string) ([]string, error) {
+	pkgbuilds := make([]string, 0, len(names))
+	var mux sync.Mutex
+	var wg sync.WaitGroup
+	var errs multierror.MultiError
+
+	makeRequest := func(full string) {
+		defer wg.Done()
+
+		dbList, dbErr := alpmHandle.SyncDBs()
+		if dbErr != nil {
+			errs.Add(dbErr)
+			return
+		}
+
+		db, name := splitDBFromName(full)
+
+		if db == "" {
+			var pkg *alpm.Package
+			for _, alpmDB := range dbList.Slice() {
```

Comment:
Fixing the array should also be done out of `makeRequest` so you pass through dbList only once (it's way bigger than targets)

Reply:
I'm not sure about this one, shouldn't dbList be only like `core`, `community` and such?

Reply:
My bad, I was distracted, I thought you were going through the `PKGCache`. the Slice() conversion could be done outside of the anonymous function.